lib/step-function/index.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const StepFunctions = require('./wrapper');
+const StepFunctionsError = require('./error');
+
+module.exports = class StepFunction {
+
+	/**
+	 * Get the params
+	 *
+	 * @static
+	 * @param {Object} param object param
+	 * @param {string} param.arn the amazon resource name (ARN) of the state machine to execute
+	 * @param {string} [param.name] the name of the execution
+	 * @param {string} [client] client code
+	 * @param {Mixed} [data] input data
+	 * @returns
+	 */
+	static getParams(arn, name, client, data) {
+
+		if(!arn || typeof arn !== 'string' || !arn.length)
+			throw new StepFunctionsError('Arn cannot be empty and must be an string.', StepFunctionsError.codes.INVALID_ARN);
+
+		if((typeof name === 'string' && !name.trim().length) || (name && typeof name !== 'string'))
+			throw new StepFunctionsError('Name cannot be empty and must be an string.', StepFunctionsError.codes.INVALID_NAME);
+
+		if((typeof client === 'string' && !client.trim().length) || (client && typeof client !== 'string'))
+			throw new StepFunctionsError('Client cannot be empty and must be an string.', StepFunctionsError.codes.INVALID_CLIENT);
+
+		if((data && typeof data !== 'object') || Array.isArray(data))
+			throw new StepFunctionsError('Data must be an object.', StepFunctionsError.codes.INVALID_DATA);
+
+		const inputData = { ...client && { session: { clientCode: client } }, ...data && { body: data } };
+
+		return {
+			stateMachineArn: arn,
+			...name && { name },
+			...Object.keys(inputData).length && { input: JSON.stringify(inputData) }
+		};
+	}
+
+	/**
+	 * Starts a state machine execution
+	 *
+	 * @static
+	 * @param {Object} param object param
+	 * @param {string} param.arn the amazon resource name (ARN) of the state machine to execute
+	 * @param {string} [param.name] the name of the execution
+	 * @param {string} [client] client code
+	 * @param {Mixed} [data] input data
+	 * @returns
+	 */
+	static async startExecution(arn, name, client, data) {
+		const params = this.getParams(arn, name, client, data);
+		return StepFunctions.startExecution(params).promise();
+	}
+
+	/**
+	 * List the state machine executions
+	 *
+	 * @param {<type>} arn The arn
+	 * @param {<type>} extraParams The extra parameters to filter
+	 * @return {Promise} The state machine executions
```

Comment:
Esta bueno decir a qué resuelve la Promise.. Algo así:

```suggestion
	 * @return {Promise<object>} The state machine execution
```

O idealmente definir el tipo de dato con un [@typedef](https://jsdoc.app/tags-typedef.html) al principio del archivo
```js
/**
* @typedef {Object} StateMachineExecution
* @property {string} executionArn The state machine execution ARN
* @property {string} stateMachineArn The state machine ARN
* // Y asi con el resto...
*/
```

```js
* @return {Promise<StateMachineExecution>} The state machine execution
```

Parece una boludes, pero después te habilita el autocomplete en varios IDEs..